### ISSUE TITLE
Spell Points

### ISSUE THREAD

**Issue by Kharmans** [foundryvtt/dnd5e#1966]
So, it would be very cool if the system supports Spell Points!

**Comment by TwoCrow** [comment]
**SRD and GMs**
I'd love to see this feature added myself. It doesn't appear to be in the SRD, but neither are other optional rules like extra critical damage or the Sanity and Honor scores, and yet those are in the sheet.

I know a few people like using spell points as an alternative to normal spell slots, as it allows you to cast a greater number of lower-level spells at higher levels, making you feel like you've mastered those spells. This would also be very useful to anyone using variant magic systems as a way to model mana, or for magic systems that work with 5e, like Spheres of Power, which rely on spell points as a mechanic.

I don't think this feature would help GMs run 5e any better, though.

**Features**
There would need to be a resource somewhere on the spell tab or character sheet that denotes how many spell points are currently available, along with a max value (that can be modified). Pressing the long rest button should refill this to max.

Additionally, there would need to be some way to specify how many spell points classes get per level. The variant rule outlines a table by which spell points are granted based on the character's level, with half casters and the like receiving an amount equal to half or a third of their level. This table also outlines the max spell level that can be cast by a character of that level.

Perhaps in place of current and max spell slots, there would be the cost of casting a spell of that level. Casting the spell should automatically decrease the number of spell points the player currently has.

Lastly, spells of 6th level or higher can only be cast once per level per long rest. This should be modeled in some way, but should be left optional in case GMs and players are using a variant magic system.

I think it would be best to enable spell points per sheet rather than for all sheets in the game, if that is possible. That way, it can be purely up to player preference to use the variant rule.

**Comment by Zanderaf** [comment]
Maybe an option to adjust when we get those points back, either on a short or long rest similar to the whole Pact Slot / Spell Slots thing.

I will +1 the whole resource being visible on somewhere in the spellbook tab, maybe something like the what the 3.0 character sheet uses for encumbrance and carry weight could be used as a some kind of visual display? Spell slots having those pips while spell points have a visual bar.

**Comment by Zanderaf** [comment]
Did a bit of a write-up in case my brain ever wants to fully work on this over the summer. Just collecting my thoughts with the whole spell point stuff since this is already the issue for it and I don't wanna make another one for it.

# **Spell Points:**

- Casting a spell with **Spell Slots** requires a spell slot to be expended in order to cast a spell. Spell slots of 1st level result in a 1st level spell, Spell slots of 2nd level in a 2nd level spell, so on and so forth.
- Casting a spell with **Spell Points** requires a certain number of points to be expended to cast a spell at a specific level. Expending 2 points will result in a spell being cast a 1st level, while 3 points will produce a spell cast at 2nd level.

There are also stipulations about casting spells of levels 6-9, with a character only being to create one of each slot above 5th level per long rest.

Having the option to have both "Spell Slots" and "Spell Points" on the same sheet via two different spellcasting progressions. (Ex. - 2 1st level Spell Slots, and 6 "Spell Points"),

# **Basic Needs:**
- Spell Points "Progression" should behave similarly to normal spellcasting progression such as "Half-Caster" or "Pact Magic". A spellcasting progression should be able to be created specifically as a "Spell Point Progression".
- Each **"Spell Point Progression"** should be able to specify a number of variable things that come into play:
     - Total Amount of "Spell Points" per level.
     - The specific amount of "Spell Points" required to cast a spell of a certain level.
     - The amount of times a character is able to "cast" a spell of a certain level. (IE the 6th-9th level spell limitation)
     - The maximum level of spell able to be "cast" using spell points. (DMG states you have 6 points at level 2, but casting a 2nd level spell costs 3 points, so math-wise you would be able to but you don't get the ability to cast 2nd level spells until level 3 assuming "Full-Caster")
- Expended amounts of "Spell Points" should be able to emulate spell slots of a certain level for items & features that utilize Consumption such as Paladin's Divine Smite feature.
- Limit the maximum level of spell a character is able to cast with these "Spell Points" through some sort of "Spell Level Limit"
- Have a place in the character sheet, either in the spell book or somewhere else to display the amount of a "Spell Points" a character has access to similar to spell slots.

# **Other Issues / Examples**

### Merging Pools
One thing I did think about is say we have a Sorcerer with "Full-Caster" type Spell Points. We take that Spell Point Progression and half it, creating a "Half-Caster" progression.

A character with levels in two classes that both utilize the same type of "Spell Points" should be able to merge the two in a singular pool just as you would when Multiclassing a Cleric with a Paladin. This would also possibly result in a higher "Spell Limit" similar to how you would sometimes be able to cast a spell of a higher level than you feasible have spells for such as when multiclassing two "Full-Casters"

### Different Spell Point Pools
Just like there are different types of "Spell Slots" in the system, different types of "Spell Points" should be able to be added in and kept separate from one another.

**Example:** A character multiclass-ed into both a Sorcerer (Spell Points) and a custom psionic class (Psi Points). Those two separate "Spell Point" pools should remain unique in their progression just as Spell Slots & Pact Slots (as well as custom progressions) remain unique between say, Wizard & Warlock.

**Comment by Kharmans** [comment]
@Zanderaf, there's a module that already do almost all of the things that you write above. Maybe the module can clarify some things for you guys. The module is **[Advanced Magic - Spell Points System 5e](https://github.com/misthero/dnd5e-spellpoints)**.

**Comment by Zanderaf** [comment]
@Kharmans I do know about the module, but as good as it is, this is still an attempt to get it in the system itself as a variant rule from the DMG. I've tried the module, and currently use it, but it does have it's problems and shortcomings. 

Most of the the things I wrote down in my previous message were my attempt at consolidating everything I've heard in the discord server for spell points. Having them be their own spellcasting progression type and able to have them tracked in a way like spell slots are as opposed to "Item" is a big part of it.

**Comment by krbz999** [comment]
we've an upcoming third party module that would make use of spell points for a new class.

i don't believe right now that the refactored spellcasting would trivialize support for this (could be wrong tho) as there are things like variable consumption (consuming X amount of points rather than just 1 spell slot of some size) etc.

**Comment by Kharmans** [comment]
Yeah, I am making use of the resources on items to cover that. The downside is that I have to configure every single item with that and the spell slots are still showing on sheet.

**Comment by Blacklands** [comment]
I would also love native system support for this; the Spell Points module is quite nice but it's limited in what it can do if it also doesn't want to mess with the character sheet rendering itself. The sheet still prominently features the spell slots, of course, and the slots still show up in the dialogue windows when casting, and so on. It's just confusing for newer players, especially. Native support could do a lot here and also would make it easier to interface with the spell points for content that specifically uses them!

Plus, having support for this would also mean that any third party content that wants to do "magic points" of any sorts (e.g. a hypothetical third-party "Psion" class that might use "Psi Points" instead of spell slots for its spellcasting) could also rely on having system support for this, instead of needing to depend on another module for it (and still have the spell slots showing anyway)!
